src/control/cmd/dmg/pool.go

```diff
@@ -204,6 +236,104 @@ func (cmd *PoolCreateCmd) Execute(args []string) error {
 	return nil
 }
 
+// XXX DAOS-9196 Some extra bytes are needed for the metadata of the pool: at least 135 MB
+// are needed.  The extraByes is a little bigger than this limit, because the size of the
+// metadata will eventually grow along the life of the pool.
+const PoolMetadataBytes uint64 = uint64(200) * uint64(humanize.MiByte)
+
+// Return the maximal size of a pool which could be created with all the storage nodes
+func (cmd *PoolCreateCmd) GetMaxPoolSize(ctx context.Context) (scmBytes uint64, nvmeBytes uint64, errOut error) {
```

Comment:
This is not the correct place for this logic, because it means that only the `dmg` tool will be able to create auto-sized pools. Instead, this should be built into `lib/control/pool.go` so that Control API users may benefit from it.

Reply:
Following tasks will be done:
- [ ] Code refactoring, move `GetMaxPoolSize` function to `lib/control/pool.go`